tests/test_cli.py

```diff
@@ -15,18 +20,317 @@ def test_header(mock_cli):
+            params["parallel-downloads"],
+        )
+
+        mock_dl.return_value.download_workflow.assert_called_once()
+
+    @mock.patch("nf_core.licences.WorkflowLicences")
+    def test_licences(self, mock_lic):
+        """Test nf-core pipeline licence is printed out and cli parameters are passed on."""
+        licence_text = "dummy licence text"
+        mock_lic.return_value.run_licences.return_value = licence_text
+
+        params = {
+            "json": None,
+        }
+
+        cmd = ["licences"] + self.assemble_params(params) + ["pipeline_name"]
+        result = self.invoke_cli(cmd)
+
+        assert result.exit_code == 0
+        assert licence_text in result.output
+
+        mock_lic.assert_called_once_with(cmd[-1])
+
+    @mock.patch("nf_core.licences.WorkflowLicences")
+    def test_licences_log_error(self, mock_lic):
+        """Test LookupError is logged"""
+        error_txt = "LookupError has been raised"
+        mock_lic.return_value.run_licences.side_effect = LookupError(error_txt)
+
+        cmd = ["licences", "pipeline_name"]
+        with self.assertLogs() as captured_logs:
+            result = self.invoke_cli(cmd)
+
+        assert result.exit_code == 1
+        assert error_txt in captured_logs.output[-1]
+        assert captured_logs.records[-1].levelname == "ERROR"
+
+    @mock.patch("nf_core.create.PipelineCreate")
+    def test_create(self, mock_create):
+        """Test nf-core pipeline is created and cli parameters are passed on."""
+        params = {
+            "name": "pipeline name",
+            "description": "pipeline description",
+            "author": "Kalle Anka",
+            "version": "1.2.3",
+            "no-git": None,
+            "force": None,
+            "outdir": "/path/outdir",
+            "template-yaml": "file.yaml",
+            "plain": None,
+        }
+
+        cmd = ["create"] + self.assemble_params(params)
+        result = self.invoke_cli(cmd)
+
+        assert result.exit_code == 0
+        mock_create.assert_called_once_with(
+            params["name"],
+            params["description"],
+            params["author"],
+            params["version"],
+            "no-git" in params,
+            "force" in params,
+            params["outdir"],
+            params["template-yaml"],
+            "plain" in params,
+        )
+        mock_create.return_value.init_pipeline.assert_called_once()
+
+    @mock.patch("nf_core.utils.is_pipeline_directory")
+    @mock.patch("nf_core.lint.run_linting")
+    def test_lint(self, mock_lint, mock_is_pipeline):
+        """Test nf-core lint"""
+        mock_lint_results = (mock.MagicMock, mock.MagicMock)
+        mock_lint_results[0].failed = []
+        mock_lint_results[1].failed = []
+        mock_lint.return_value = mock_lint_results
+
+        temp_pipeline_dir = tempfile.NamedTemporaryFile()
+        params = {
+            "dir": temp_pipeline_dir.name,
+            "release": None,
+            "fix": "fix test",
```

Comment:
shouldn't this be `none`?

Reply:
Doesn't `fix` take a string as parameter? Documentation says "Attempt to automatically fix specified lint test". Or maybe you're thinking about `nf-core modules lint --fix-version` instead of `nf-core lint --fix test`?

Reply:
sorry, you are correct. Misremembered it probably from a different command/tool 🙃